funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -755,18 +791,35 @@ def start(self, poll_period=None):
 
             self.total_pending_task_count -= dispatched_task
 
+            # Task cancel is high priority, so we'll process all requests
+            # in one go
+            while True:
+                try:
+                    manager, task_id = self.task_cancel_running_queue.get(block=False)
+                    logger.debug(f"[MAIN] Task:{task_id} on manager:{manager} is now CANCELLED while running")
+                    cancel_message = pickle.dumps(('TASK_CANCEL', task_id))
+                    self.task_outgoing.send_multipart([manager, b'', cancel_message])
+
+                except queue.Empty:
+                    break
+
             for manager in task_dispatch:
                 tasks = task_dispatch[manager]
                 if tasks:
                     logger.info("[MAIN] Sending task message {} to manager {}".format(tasks, manager))
-                    serializd_raw_tasks_buffer = pickle.dumps(tasks)
-                    # self.task_outgoing.send_multipart([manager, b'', pickle.dumps(tasks)])
-                    self.task_outgoing.send_multipart([manager, b'', serializd_raw_tasks_buffer])
+                    serialised_raw_tasks_buffer = pickle.dumps(tasks)
+                    self.task_outgoing.send_multipart([manager, b'', serialised_raw_tasks_buffer])
 
                     for task in tasks:
                         task_id = task["task_id"]
-                        logger.debug(f"[MAIN] Task {task_id} is now WAITING_FOR_LAUNCH")
-                        self.task_status_deltas[task_id] = TaskStatusCode.WAITING_FOR_LAUNCH
+                        if self.task_cancel_pending_trap and task_id in self.task_cancel_pending_trap:
```

Comment:
this if is the same as just `if task_id in self.task_cancel_pending_trap` right? the first clause isn't catching anything else?